

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Reference &mdash; nimgen  documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/gallery.css" type="text/css" />
  <link rel="stylesheet" href="_static/gallery-binder.css" type="text/css" />
  <link rel="stylesheet" href="_static/gallery-dataframe.css" type="text/css" />
  <link rel="stylesheet" href="_static/gallery-rendered-html.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Analyse gene expressions in webgestalt" href="auto_examples/run_get_webgestalt.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> nimgen
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installing nimgen</a></li>
<li class="toctree-l1"><a class="reference internal" href="auto_examples/index.html">Examples</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Reference</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#module-nimgen.expressions">Main functions</a></li>
</ul>
</li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">nimgen</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Reference</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/api.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="reference">
<h1>Reference<a class="headerlink" href="#reference" title="Permalink to this headline">¶</a></h1>
<div class="section" id="module-nimgen.expressions">
<span id="main-functions"></span><h2>Main functions<a class="headerlink" href="#module-nimgen.expressions" title="Permalink to this headline">¶</a></h2>
<dl class="py function">
<dt id="nimgen.expressions.get_gene_expression">
<code class="sig-prename descclassname"><span class="pre">nimgen.expressions.</span></code><code class="sig-name descname"><span class="pre">get_gene_expression</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">weights</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atlas</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">allen_data_dir</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_expressions</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">force_recompute</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">multiple_correction</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'fdr_bh'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#nimgen.expressions.get_gene_expression" title="Permalink to this definition">¶</a></dt>
<dd><p>Get the genes expressed in the atlas that correlate with the
specified weights.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>weights</strong><span class="classifier">list(float) or np.ndarray or pandas.Series or pandas.DataFrame</span></dt><dd><p>Weights for each ROI in the atlas</p>
</dd>
<dt><strong>atlas</strong><span class="classifier">niimg-like object</span></dt><dd><p>A parcellation image in MNI space, where each parcel is identified by a
unique integer ID.</p>
</dd>
<dt><strong>allen_data_dir</strong><span class="classifier">pathlib.Path or string</span></dt><dd><p>Directory where expression data should be downloaded (if it does not
already exist) / loaded.</p>
</dd>
<dt><strong>save_expressions</strong><span class="classifier">bool</span></dt><dd><p>If True (default), the expressions on the atlas will be saved for later
calls to the function with the same atlas. It will create a file named
<cite>nimgen_{atlas}_expressions.csv</cite> next to the atlas file.</p>
</dd>
<dt><strong>force_recompute</strong><span class="classifier">bool</span></dt><dd><p>If True, disregard the previously saved expressions and recompute the
gene expressions using abagen. Defaults to False.</p>
</dd>
<dt><strong>multiple_correction</strong><span class="classifier">str</span></dt><dd><p>Method to use for the correction for multiple comparisons. Check
<cite>statsmodels.stats.multitest.mutipletests</cite> for a list of available
options. Defaults to ‘fdr_bh’</p>
</dd>
<dt><strong>alpha</strong><span class="classifier">float</span></dt><dd><p>FWER, family-wise error rate, e.g. 0.1. Defaults to 0.05.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>genes</strong><span class="classifier">list(str)</span></dt><dd><p>A list will all the genes that are expressed in the atlas and correlate
with the weights.</p>
</dd>
</dl>
</dd>
<dt class="field-odd">Raises</dt>
<dd class="field-odd"><dl class="simple">
<dt>ValueError</dt><dd><p>If there is a problem with the Atlas and/or the weights.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<span class="target" id="module-nimgen.webgestalt"></span><dl class="py function">
<dt id="nimgen.webgestalt.webgestalt">
<code class="sig-prename descclassname"><span class="pre">nimgen.webgestalt.</span></code><code class="sig-name descname"><span class="pre">webgestalt</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">genes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">enrich_db_category</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">enrich_db_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">enrich_method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'ORA'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">id_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'genesymbol'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">refset</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'agilent_sureprint_g3_ge_8x60k'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">organism</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'hsapiens'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sig_method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'fdr'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sig_value</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">browser</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'chrome'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#nimgen.webgestalt.webgestalt" title="Permalink to this definition">¶</a></dt>
<dd><p>Set-up an analysis to be performed in webgestalt.
See <a class="reference external" href="http://www.webgestalt.org">http://www.webgestalt.org</a></p>
<p>Requires selenium and the browser driver installed. 
See <a class="reference external" href="https://juaml.github.io/nimgen/installation.html">https://juaml.github.io/nimgen/installation.html</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>genes</strong><span class="classifier">list(str)</span></dt><dd><p>A list will all the genes that will take part in the webgestalt
analysis.</p>
</dd>
<dt><strong>enrich_db_category</strong><span class="classifier">str</span></dt><dd><p>Functional database category. Options are geneontology, pathway,
network, disease, drug, phenotype, chromosomalLocation,
community-contributed, others.</p>
</dd>
<dt><strong>enrich_db_name</strong><span class="classifier">str</span></dt><dd><p>Depends on enrich_db_category. Check <a class="reference external" href="http://www.webgestalt.org">http://www.webgestalt.org</a> for
options.</p>
</dd>
<dt><strong>enrich_method</strong><span class="classifier">str</span></dt><dd><p>Options are: ORA, GSEA, NTA. Defaults to ‘ORA’.</p>
</dd>
<dt><strong>id_type: str</strong></dt><dd><p>Gene ID Type of the gene list. Check <a class="reference external" href="http://www.webgestalt.org">http://www.webgestalt.org</a> for
options. Defaulkts to ‘genesymbol’.</p>
</dd>
<dt><strong>refset</strong><span class="classifier">str</span></dt><dd><p>Built-in reference set or microarray platform (only when
<cite>enrich_method=’ORA’</cite>). Defaults to ‘agilent_sureprint_g3_ge_8x60k’.</p>
</dd>
<dt><strong>organism</strong><span class="classifier">str</span></dt><dd><p>Organism of interest. Options are athaliana, btaurus, celegans,
cfamiliaris, dmelanogaster, drerio, ggallus, hsapiens, mmusculus,
rnorvegicus, scerevisiae, sscrofa, others. Defaults to ‘hsapiens’.</p>
</dd>
<dt><strong>sig_method</strong><span class="classifier">str</span></dt><dd><p>Methods to use for identifying the enriched categories. ‘fdr’ means the
enriched categories will be identified based on FDR threshold and
‘top’ means the categories will be first ranked based on the FDR and
then the top most significant categories will be selected. Defaults to
‘fdr’.</p>
</dd>
<dt><strong>sig_value</strong><span class="classifier">numerical</span></dt><dd><p>Threshold for the significance method. Defaults to 0.05.</p>
</dd>
<dt><strong>browser</strong><span class="classifier">str</span></dt><dd><p>Browser to use. Supported options are ‘chrome’ (default) and ‘firefox’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Raises</dt>
<dd class="field-even"><dl class="simple">
<dt>ValueError</dt><dd><p>If there is a problem with the browser and/or selenium</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="auto_examples/run_get_webgestalt.html" class="btn btn-neutral float-left" title="Analyse gene expressions in webgestalt" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2020, Authors of nimgen.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>